modules/core/src/lib/layer.ts

```diff
@@ -468,8 +468,13 @@ export default abstract class Layer<PropsT = {}> extends Component<PropsT & Requ
 
     const {props, oldProps} = params;
     const neededPickingBuffer =
-      Number.isInteger(oldProps.highlightedObjectIndex) || oldProps.pickable;
-    const needPickingBuffer = Number.isInteger(props.highlightedObjectIndex) || props.pickable;
+      Number.isInteger(oldProps.highlightedObjectIndex) ||
+      oldProps.pickable ||
+      oldProps.extensions?.some(e => e.needsPickingBuffer());
```

Comment:
Not blocking, but I don't think this will work - if `extension.needsPickingBuffer()` accesses layer props, it will always end up reading the current value, not the previous value. `needsPickingBuffer` may have to be cached in `internalState`.